### DISCUSSION TITLE
Recommended way of getting prod/deploy DB on to local

### DISCUSSION THREAD

**Discussion by sasacocic** [supabase/supabase#3581] [Questions] [answered]
I started using supabase in prod immediately, which is fine because I didn't have any users and I just wanted to try it out.

Now I have some users, and I'd like to setup local development so I don't mess anything up. I've got a local instance running, and that was easy enough to do.

What isn't as straight forward is figuring out how I can make my local database match what I currently have in prod. What is the recommended way I do that?

e.g. Lets say I have 2 tables in prod. One called `people` and it has 1 column name, and another called `cities` with 1 column `size`

`people`

| name |
| ---- |
| josh |
| randy |
| amanda |

`cities`

| size |
| ---- |
| 120 |
| 200 |
| .... |

I don't really care about the data I just want the schemas to be the same

EDIT:

I've tried `supabase link` , `supabase db restore`, and some of these. But these all seem to go from local to deploy, but the other way around doesn't seem possible.

EDIT 2:

This question basically boils down to this question: https://github.com/supabase/supabase/discussions/2307#discussion-3453863

which looks like it's really old 😔

**Comment by sasacocic** [comment, marked as answer]
I just wanted to answer my own question. After messing around with the supabase cli and reading docs randomly (which btw didn't clearly answer this question for me) it seems that supabase does this automatically. If you just

1. `supabase init`
2. `supabase link --url <<< 'postgresql://postgres:<your_password>@db.<your_project_ref>.supabase.co:5432/postgres'`
3. `supabase start`
4. `supabase db restore` - this might be unnecessary


At first I did some random permutation of this. I added a new table, and tried to see if the table would be picked up by the cli, which didn't seem to happen. Don't know if I need to nuke the local supabse dir. and try again, but supabase also makes a 'migrations' table on the "prod" database. Not sure if that needs to be nuked as well. I'm gonna mess around with it and see what happens. Might report back might not 🤷‍♂️

**Reply by soedirgo** [reply]
Yes, `supabase link` will do this for you (it creates a new migration locally). I've outlined this in the [tour](https://github.com/supabase/cli/tree/main/examples/tour), but I'll need to link that in the README to make it more prominent. You won't need `supabase db restore` since `supabase start` would have captured that new migration at that point.

If you want to start from a clean slate, other than nuking the `supabase` dir, you'll also need to drop the migrations schema: `DROP SCHEMA supabase_migrations CASCADE;`.

If you have other issues, keep up posted 👍
